Failed to load payload for github-repo: timedout

**Description**

Running the plugin against big monorepos with many dependency files results in timeout from one of the GraphQl queries . The the [dependencyGraphManifests](https://github.com/revanite-io/pvtr-github-repo/blob/eff5bcb63b70227b76ca41a54d834164eafebea9/data/graphql-data.go#L92) part in [GraphqlRepoData](https://github.com/revanite-io/pvtr-github-repo/blob/eff5bcb63b70227b76ca41a54d834164eafebea9/data/graphql-data.go#L13-L14) causes the whole query to time out

**Configuration**
```
loglevel: trace
write-directory: evaluation_results
write: true
output: json
services:
  crowd.dev:
    plugin: github-repo

    policy:
      catalogs:
        - OSPS_B
      applicability:
        - Maturity Level 1

    vars:
      owner: CrowdDotDev
      repo: crowd.dev
      token: <redacted>
```

**Output**
```
./privateer run
2025/04/29 15:24:50 vars: map[]
2025-04-29T15:24:56.493+0200 [ERROR] tests failed in plugin crowd.dev: Failed to load payload for github-repo: timedout
```

**Expected Behavior**
The plugin should be creating crowd.dev.json under `evaluation_results/crowd.dev`

**Actual Behavior**
Log file [crowd.dev.log](https://github.com/user-attachments/files/19959985/crowd.dev.log) is created, but json (`crowd.dev.json`) is missing

Confirmed this with a debug run against `rust-lang/cargo` as part of #76 

<details>
<summary>
Failed run: 

`go run . debug --service=cargo`

</summary>

```sh
Running in debug mode
2025/04/30 14:10:41 vars: map[owner:rust-lang repo:cargo token:<redacted>]
2025-04-30T14:10:51.571-0500 [ERROR] Error querying GitHub GraphQL API: non-200 OK status code: 502 Bad Gateway body: "{\n   \"data\": null,\n   \"errors\":[\n      {\n         \"message\":\"Something went wrong while executing your query on 2025-04-30T19:10:51+00:00. This may be the result of a timeout, or it could be a GitHub bug. Please include `E0B5:19B9F7:17D7181:2F7C270:681275B1` when reporting this issue.\"\n      }\n   ]\n}\n"
failed to load payload for github-repo: non-200 OK status code: 502 Bad Gateway body: "{\n   \"data\": null,\n   \"errors\":[\n      {\n         \"message\":\"Something went wrong while executing your query on 2025-04-30T19:10:51+00:00. This may be the result of a timeout, or it could be a GitHub bug. Please include `E0B5:19B9F7:17D7181:2F7C270:681275B1` when reporting this issue.\"\n      }\n   ]\n}\n"
```
</details>

--- a/data/graphql-data.go
+++ b/data/graphql-data.go
@@ -1,14 +1,5 @@
 package data
 
-import (
-	"context"
-	"fmt"
-	"path/filepath"
-
-	"github.com/privateerproj/privateer-sdk/config"
-	"github.com/shurcooL/githubv4"
-)
-
 // GraphqlRepoData is used in a query to get general repository information
 type GraphqlRepoData struct {
 	Repository struct {
@@ -85,22 +76,8 @@ type GraphqlRepoData struct {
 			Description string
 		}
 		ContributingGuidelines struct {
-			Body         string
-			ResourcePath string
+			Body string
 		}
-		DependencyGraphManifests struct {
-			TotalCount int
-			Nodes      []struct {
-				Filename     string
-				Dependencies struct {
-					TotalCount int
-					Nodes      []struct {
-						PackageName  string
-						Requirements string
-					}
-				} `graphql:"dependencies(first: 100)"`
-			} `graphql:"nodes"`
-		} `graphql:"dependencyGraphManifests(first: 100)"`
 		Releases struct {
 			Nodes []struct {
 				TagName string
@@ -114,114 +91,4 @@ type GraphqlRepoData struct {
 			}
 		} `graphql:"releases(first: 1, orderBy: {field: CREATED_AT, direction: DESC})"`
 	} `graphql:"repository(owner: $owner, name: $name)"`
-}
-
-// GraphqlRepoTree is used in a query to get top 3 levels of the repository contents
-type GraphqlRepoTree struct {
-	Repository struct {
-		Object struct {
-			Tree struct {
-				Entries []struct {
-					Name   string
-					Type   string // "blob" for files, "tree" for directories
-					Path   string
-					Object *struct {
-						Tree struct {
-							Entries []struct {
-								Name   string
-								Type   string
-								Path   string
-								Object *struct {
-									Tree struct {
-										Entries []struct {
-											Name string
-											Type string
-											Path string
-										}
-									} `graphql:"... on Tree"`
-								} `graphql:"object"`
-							}
-						} `graphql:"... on Tree"`
-					} `graphql:"object"`
-				}
-			} `graphql:"... on Tree"`
-		} `graphql:"object(expression: $branch)"`
-	} `graphql:"repository(owner: $owner, name: $name)"`
-}
-
-func checkTreeForBinaries(tree *GraphqlRepoTree, binariesFound []string) []string {
-	for _, entry := range tree.Repository.Object.Tree.Entries {
-		binariesFound = identifyBinaries(binariesFound, entry.Type, entry.Name)
-		if entry.Type == "tree" {
-			for _, subEntry := range entry.Object.Tree.Entries {
-				binariesFound = identifyBinaries(binariesFound, subEntry.Type, subEntry.Name)
-				if subEntry.Type == "tree" {
-					for _, subSubEntry := range subEntry.Object.Tree.Entries {
-						binariesFound = identifyBinaries(binariesFound, subSubEntry.Type, subSubEntry.Name)
-						// if subSubEntry.Type == "tree" {
-						// TODO: The current GraphQL call stops after 3 levels of depth.
-						// Additional API calls will be required for recursion if another tree is found.
-						// }
-					}
-				}
-			}
-		}
-	}
-	return binariesFound
-}
-
-func identifyBinaries(binariesFound []string, filetype string, filename string) []string {
-	if filetype == "blob" {
-		if isBinaryFile(filename) {
-			binariesFound = append(binariesFound, filename)
-		}
-	}
-	return binariesFound
-}
-
-// TODO: this is a lightweight check, looking at filenames only.
-// GitHub's GraphQL API has an 'isBinary' field that could be used for a more accurate check,
-// but I didn't manage to get that query working as expected.
-func isBinaryFile(filename string) bool {
-	binaryExtensions := map[string]bool{
-		"": true, ".exe": true, ".dll": true, ".so": true, ".pdf": true,
-		".zip": true, ".tar": true, ".mp4": true, ".mp3": true,
-	}
-	knownFilenames := map[string]bool{
-		// Extend this with more known filenames as needed
-		"README": true, "LICENSE": true, "CHANGELOG": true, "CONTRIBUTING": true,
-		"CODE_OF_CONDUCT": true, "TODO": true, "SECURITY": true, "NOTICE": true, "CODEOWNERS": true,
-		".gitignore": true, ".gitattributes": true, "Makefile": true, "Dockerfile": true,
-		"Vagrantfile": true, "Gemfile": true, "Procfile": true, "Brewfile": true, "MANIFEST": true,
-	}
-	if knownFilenames[filename] {
-		return false
-	}
-	ext := filepath.Ext(filename)
-	return binaryExtensions[ext]
-}
-
-func fetchGraphqlRepoTree(config *config.Config, client *githubv4.Client, branch string) (tree *GraphqlRepoTree, err error) {
-	path := "" // TODO: I suspected we should be able to target subdirectories this way, but it hasn't succeeded
-
-	fullPath := fmt.Sprintf("%s:%s", branch, path) // Ensure correct format
-
-	variables := map[string]interface{}{
-		"owner":  githubv4.String(config.GetString("owner")),
-		"name":   githubv4.String(config.GetString("repo")),
-		"branch": githubv4.String(fullPath),
-	}
-
-	err = client.Query(context.Background(), &tree, variables)
-
-	return tree, err
-}
-
-func getSuspectedBinaries(client *githubv4.Client, config *config.Config, branchName string) (suspectedBinaries []string, err error) {
-	tree, err := fetchGraphqlRepoTree(config, client, branchName)
-	if err != nil {
-		return nil, err
-	}
-	binaryFileNames := checkTreeForBinaries(tree, []string{})
-	return binaryFileNames, nil
 }

--- a/data/graphql-dependency-manifests.go
+++ b/data/graphql-dependency-manifests.go
@@ -1,0 +1,41 @@
+package data
+
+import (
+	"context"
+
+	"github.com/privateerproj/privateer-sdk/config"
+	"github.com/shurcooL/githubv4"
+)
+
+type DependencyManifestsPage struct {
+	Repository struct {
+		DependencyGraphManifests struct {
+			TotalCount int
+		}
+	} `graphql:"repository(owner: $owner, name: $name)"`
+}
+
+type ManifestNode struct {
+	Filename     string
+	Dependencies []Dependency
+}
+
+type Dependency struct {
+	PackageName  string
+	Requirements string
+}
+
+func countDependencyManifests(client *githubv4.Client, cfg *config.Config) (int, error) {
+	var query DependencyManifestsPage
+	variables := map[string]interface{}{
+		"owner": githubv4.String(cfg.GetString("owner")),
+		"name":  githubv4.String(cfg.GetString("repo")),
+	}
+
+	err := client.Query(context.Background(), &query, variables)
+	if err != nil {
+		return 0, err
+	}
+
+	return query.Repository.DependencyGraphManifests.TotalCount, nil
+}

--- a/data/graphql-repo-tree.go
+++ b/data/graphql-repo-tree.go
@@ -1,0 +1,120 @@
+package data
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/privateerproj/privateer-sdk/config"
+	"github.com/shurcooL/githubv4"
+)
+
+// GraphqlRepoTree is used in a query to get top 3 levels of the repository contents
+type GraphqlRepoTree struct {
+	Repository struct {
+		Object struct {
+			Tree struct {
+				Entries []struct {
+					Name   string
+					Type   string // "blob" for files, "tree" for directories
+					Path   string
+					Object *struct {
+						Tree struct {
+							Entries []struct {
+								Name   string
+								Type   string
+								Path   string
+								Object *struct {
+									Tree struct {
+										Entries []struct {
+											Name string
+											Type string
+											Path string
+										}
+									} `graphql:"... on Tree"`
+								} `graphql:"object"`
+							}
+						} `graphql:"... on Tree"`
+					} `graphql:"object"`
+				}
+			} `graphql:"... on Tree"`
+		} `graphql:"object(expression: $branch)"`
+	} `graphql:"repository(owner: $owner, name: $name)"`
+}
+
+func checkTreeForBinaries(tree *GraphqlRepoTree, binariesFound []string) []string {
+	for _, entry := range tree.Repository.Object.Tree.Entries {
+		binariesFound = identifyBinaries(binariesFound, entry.Type, entry.Name)
+		if entry.Type == "tree" {
+			for _, subEntry := range entry.Object.Tree.Entries {
+				binariesFound = identifyBinaries(binariesFound, subEntry.Type, subEntry.Name)
+				if subEntry.Type == "tree" {
+					for _, subSubEntry := range subEntry.Object.Tree.Entries {
+						binariesFound = identifyBinaries(binariesFound, subSubEntry.Type, subSubEntry.Name)
+						// if subSubEntry.Type == "tree" {
+						// TODO: The current GraphQL call stops after 3 levels of depth.
+						// Additional API calls will be required for recursion if another tree is found.
+						// }
+					}
+				}
+			}
+		}
+	}
+	return binariesFound
+}
+
+func identifyBinaries(binariesFound []string, filetype string, filename string) []string {
+	if filetype == "blob" {
+		if isBinaryFile(filename) {
+			binariesFound = append(binariesFound, filename)
+		}
+	}
+	return binariesFound
+}
+
+// TODO: this is a lightweight check, looking at filenames only.
+// GitHub's GraphQL API has an 'isBinary' field that could be used for a more accurate check,
+// but I didn't manage to get that query working as expected.
+func isBinaryFile(filename string) bool {
+	binaryExtensions := map[string]bool{
+		"": true, ".exe": true, ".dll": true, ".so": true, ".pdf": true,
+		".zip": true, ".tar": true, ".mp4": true, ".mp3": true,
+	}
+	knownFilenames := map[string]bool{
+		// Extend this with more known filenames as needed
+		"README": true, "LICENSE": true, "CHANGELOG": true, "CONTRIBUTING": true,
+		"CODE_OF_CONDUCT": true, "TODO": true, "SECURITY": true, "NOTICE": true, "CODEOWNERS": true,
+		".gitignore": true, ".gitattributes": true, "Makefile": true, "Dockerfile": true,
+		"Vagrantfile": true, "Gemfile": true, "Procfile": true, "Brewfile": true, "MANIFEST": true,
+	}
+	if knownFilenames[filename] {
+		return false
+	}
+	ext := filepath.Ext(filename)
+	return binaryExtensions[ext]
+}
+
+func fetchGraphqlRepoTree(config *config.Config, client *githubv4.Client, branch string) (tree *GraphqlRepoTree, err error) {
+	path := "" // TODO: I suspected we should be able to target subdirectories this way, but it hasn't succeeded
+
+	fullPath := fmt.Sprintf("%s:%s", branch, path) // Ensure correct format
+
+	variables := map[string]interface{}{
+		"owner":  githubv4.String(config.GetString("owner")),
+		"name":   githubv4.String(config.GetString("repo")),
+		"branch": githubv4.String(fullPath),
+	}
+
+	err = client.Query(context.Background(), &tree, variables)
+
+	return tree, err
+}
+
+func getSuspectedBinaries(client *githubv4.Client, config *config.Config, branchName string) (suspectedBinaries []string, err error) {
+	tree, err := fetchGraphqlRepoTree(config, client, branchName)
+	if err != nil {
+		return nil, err
+	}
+	binaryFileNames := checkTreeForBinaries(tree, []string{})
+	return binaryFileNames, nil
+}

--- a/data/payload.go
+++ b/data/payload.go
@@ -13,9 +13,10 @@ import (
 type Payload struct {
 	*GraphqlRepoData
 	*RestData
-	Config             *config.Config
-	SuspectedBinaries  []string
-	RepositoryMetadata RepositoryMetadata
+	Config                   *config.Config
+	SuspectedBinaries        []string
+	RepositoryMetadata       RepositoryMetadata
+	DependencyManifestsCount int
 }
 
 func Loader(config *config.Config) (payload interface{}, err error) {
@@ -34,17 +35,22 @@ func Loader(config *config.Config) (payload interface{}, err error) {
 	if err != nil {
 		return nil, err
 	}
+	dependencyManifestsCount, err := countDependencyManifests(client, config)
+	if err != nil {
+		return nil, err
+	}
 
 	rest, err := getRestData(ghClient, config)
 	if err != nil {
 		return nil, err
 	}
 	return interface{}(Payload{
-		GraphqlRepoData:    graphql,
-		RestData:           rest,
-		Config:             config,
-		SuspectedBinaries:  suspectedBinaries,
-		RepositoryMetadata: repositoryMetadata,
+		GraphqlRepoData:          graphql,
+		RestData:                 rest,
+		Config:                   config,
+		SuspectedBinaries:        suspectedBinaries,
+		RepositoryMetadata:       repositoryMetadata,
+		DependencyManifestsCount: dependencyManifestsCount,
 	}), nil
 }
 

--- a/evaluation_plans/osps/quality/steps.go
+++ b/evaluation_plans/osps/quality/steps.go
@@ -2,7 +2,6 @@ package quality
 
 import (
 	"fmt"
-	"path/filepath"
 	"strings"
 
 	"github.com/revanite-io/pvtr-github-repo/evaluation_plans/reusable_steps"
@@ -201,116 +200,21 @@ func verifyDependencyManagement(payloadData interface{}, _ map[string]*layer4.Ch
 	}
 
 	// Check dependency manifests
-	result, message = verifyDependencyManifests(data)
-	if result != layer4.Passed {
-		return result, message
-	}
-
-	return layer4.Passed, "Dependency management files and SBOM requirements met"
+	// TODO: Do a quality check on the dependency manifests
+	return countDependencyManifests(data)
 }
 
-type DependencyManifest struct {
-	Filename string
-	Language string
-}
-
-// Known dependency management files for various languages
-var knownManifests = []DependencyManifest{
-	{"go.mod", "Go"},
-	{"package.json", "JavaScript/Node.js"},
-	{"pom.xml", "Java (Maven)"},
-	{"build.gradle", "Java (Gradle)"},
-	{"requirements.txt", "Python"},
-	{"Pipfile", "Python (Pipenv)"},
-	{"pyproject.toml", "Python (Poetry)"},
-	{"Gemfile", "Ruby"},
-	{"composer.json", "PHP"},
-	{"Cargo.toml", "Rust"},
-	{"*.csproj", ".NET"},
-	{"mix.exs", "Elixir"},
-}
-
-// ManifestResult stores the validation result for a manifest
-type ManifestResult struct {
-	Found    bool
-	Language string
-	HasDeps  bool
-	Manifest string
-}
-
-func isManifestFile(filename string, pattern string) bool {
-	if strings.Contains(pattern, "*") {
-		matched, _ := filepath.Match(pattern, filename)
-		return matched
-	}
-	return filename == pattern
-}
-
-func verifyDependencyManifests(payloadData interface{}) (layer4.Result, string) {
+func countDependencyManifests(payloadData interface{}) (result layer4.Result, message string) {
 	data, message := reusable_steps.VerifyPayload(payloadData)
 	if message != "" {
 		return layer4.Unknown, message
 	}
 
-	if data.Repository.Object.Tree.Entries == nil {
-		return layer4.Unknown, "Repository tree entries not available"
+	manifestsCount := data.DependencyManifestsCount
+	if manifestsCount > 0 {
+		return layer4.Passed, fmt.Sprintf("Found %d dependency manifests from GitHub API", manifestsCount)
 	}
-
-	entries := data.Repository.Object.Tree.Entries
-	manifests := data.Repository.DependencyGraphManifests.Nodes
-
-	foundManifests := make(map[string]ManifestResult)
-	manifestErrors := []string{}
-
-	// Check repository contents for dependency files
-	for _, entry := range entries {
-		for _, manifest := range knownManifests {
-			if isManifestFile(entry.Name, manifest.Filename) {
-				result := ManifestResult{
-					Found:    true,
-					Language: manifest.Language,
-					Manifest: entry.Name,
-				}
-				foundManifests[manifest.Language] = result
-			}
-		}
-	}
-
-	if len(manifestErrors) > 0 {
-		return layer4.Failed, fmt.Sprintf("Dependency management issues found:\n%s",
-			strings.Join(manifestErrors, "\n"))
-	}
-
-	if len(foundManifests) == 0 {
-		return layer4.Failed, "No dependency management files found"
-	}
-
-	for _, manifest := range manifests {
-		if manifest.Dependencies.TotalCount > 0 {
-			for lang, result := range foundManifests {
-				if manifest.Filename == result.Manifest {
-					result.HasDeps = true
-					foundManifests[lang] = result
-					break
-				}
-			}
-		}
-	}
-
-	missingDeps := []string{}
-	for lang, result := range foundManifests {
-		if !result.HasDeps {
-			missingDeps = append(missingDeps,
-				fmt.Sprintf("%s (%s)", result.Manifest, lang))
-		}
-	}
-
-	if len(missingDeps) > 0 {
-		return layer4.Failed, fmt.Sprintf("No dependencies declared in: %s",
-			strings.Join(missingDeps, ", "))
-	}
-
-	return layer4.Passed, "Dependency management files present and properly configured"
+	return layer4.Failed, "No dependency manifests found in the repository by the GitHub API"
 }
 
 func documentsTestExecution(payloadData interface{}, _ map[string]*layer4.Change) (result layer4.Result, message string) {

